cranelift/codegen/src/ir/dfg.rs

```diff
@@ -1472,12 +1491,16 @@ impl DataFlowGraph {
 pub struct BlockData {
     /// List of parameters to this block.
     params: ValueList,
+
+    /// Block name.
+    name: Option<String>,
```

Comment:
Putting this into a separate `SecondaryMap<Block, String>` would reduce memory usage when the block name functionality isn't used and remove unnecessary padding and improve cpu cache efficiency when it is used.

Reply:
Also you can represent no name being given as an empty string.